torchvision/datasets/flowers102.py

```diff
@@ -0,0 +1,116 @@
+from pathlib import Path
+from typing import Any, Tuple, Callable, Optional
+
+import numpy as np
+import PIL.Image
+
+from .utils import verify_str_arg, download_and_extract_archive, download_url
+from .vision import VisionDataset
+
+
+class Flowers102(VisionDataset):
+    """`Oxford 102 Flower <https://www.robots.ox.ac.uk/~vgg/data/flowers/102/>`_ Dataset.
+
+    .. warning::
+
+        This class needs `scipy <https://docs.scipy.org/doc/>`_ to load target files from `.mat` format.
+
+    Oxford 102 Flower is an image classification dataset consisting of 102 flower categories. The
+    flowers chosen to be flower commonly occurring in the United Kingdom. Each class consists of
+    between 40 and 258 images.
+
+    The images have large scale, pose and light variations. In addition, there are categories that
+    have large variations within the category and several very similar categories.
+
+    Args:
+        root (string): Root directory of the dataset.
+        split (string, optional): The dataset split, supports ``"train"`` (default), ``"val"``, or ``"test"``.
+        transform (callable, optional): A function/transform that takes in an PIL image and returns a
+            transformed version. E.g, ``transforms.RandomCrop``.
+        target_transform (callable, optional): A function/transform that takes in the target and transforms it.
+    """
+
+    def __init__(
+        self,
+        root: str,
+        split: str = "train",
+        download: bool = True,
+        transform: Optional[Callable] = None,
+        target_transform: Optional[Callable] = None,
+    ) -> None:
+        super().__init__(root, transform=transform, target_transform=target_transform)
+        self._split = verify_str_arg(split, "split", ("train", "valid", "test"))
+        self._base_folder = Path(self.root) / "flowers-102"
+        self._meta_folder = self._base_folder / "labels"
```

Comment:
Do we actually need this? The images come in their separate folder anyway, so I feel like we can place the two other files in `self.root`.

Reply:
Hi @pmeier , The reason for creating this directory is to make it easier to determine if the two files were downloaded correctly.

Reply:
Given that we are talking about two files vs. one folder, I think we can simply check for the existence of both files? [`torchvision.datasets.utils.check_integrity`](https://github.com/pytorch/vision/blob/8265469b2526e8d520e2e5ede71454ac01e357aa/torchvision/datasets/utils.py#L66) was made for that.

Reply:
Hi @pmeier , I've updated these with `check_integrity`.